src/components/Terminal/Terminal.js

```diff
@@ -23,10 +24,24 @@ const Terminal = ({ address, modelName }) => {
     return cleanUpTerminal(terminalInstance);
   }, [address, macaroons, modelName]);
 
+  const [terminalVisible, setTerminalVisible] = useState(false);
+
   return (
-    <div className="terminal">
-      <div className="terminal-header">Juju Terminal</div>
-      <div className="terminal-shell" ref={terminalElement}></div>
+    <div
+      className={classNames("p-terminal", {
+        "is-visible": terminalVisible
+      })}
+    >
+      <div
+        className="p-terminal__header"
+        onClick={() => setTerminalVisible(!terminalVisible)}
```

Comment:
With these changes to the terminal it never displays anything but `connecting...` it still connects, returns the data, and accepts input but doesn't show that.

It's possible that it's doing this because it's hidden on the initial load so on each 'open' we may have to call `fit()` on the terminal instance  but that's just an idea. To do that here would require a bit of a code change to make it available outside of the `useEffect` - You could store it in a `useRef` to gain access to it. 

https://github.com/canonical-web-and-design/jaas-dashboard/blob/master/src/components/Terminal/setup-terminal.js#L100

If you find that this is a difficult or time consuming fix you could pull this terminal stuff out into a separate PR and we can work/land it separately.

Reply:
Works for me?

Video screen capture: 

https://cl.ly/38a02b9972ca/Screen%20Recording%202019-10-24%20at%2001.04%20pm.mov